projects/js-packages/publicize-components/src/hooks/use-media-restrictions/index.js

```diff
@@ -47,6 +47,10 @@ export const getAllowedMediaTypes = enabledConnections => {
 			: DEFAULT_RESTRICTIONS.allowedMediaTypes
 	);
 
+	if ( enabledConnections.length === 0 ) {
+		return DEFAULT_RESTRICTIONS.allowedMediaTypes;
+	}
+
 	if ( typeArrays.length === 0 ) {
 		return [];
```

Comment:
Would this change have the same effect?

```suggestion
		return DEFAULT_RESTRICTIONS.allowedMediaTypes;
```